### PR TITLE
Setup prow for kfserving

### PR DESCRIPTION
**Pull request by hougangliu**
Per #46, this PR is only for prow job to run`make test` and build docker image, we do not have e2e tests yet but like to start with unit tests to verify PRs sending in.

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/74)

### CODE REVIEW

**Comment by hougangliu** [comment]
I will take over prow setup from @yuzisun in this PR, a coming commit for prow setup

**Comment by animeshsingh** [comment]
Thanks @hougangliu for this. qq for you and @yuzisun - given we are moving to Kustomize for next release, what`s the point of doing things vis a vis ksonnet?

**Comment by yuzisun** [comment]
> Thanks @hougangliu for this. qq for you and @yuzisun - given we are moving to Kustomize for next release, what`s the point of doing things vis a vis ksonnet?

That's per discussion with @jlewi, there is no good alternative right now for testing, the option is to either use ksonnet or work on replacement. I do not think we would have time for the later.

**Comment by yuzisun** [comment]
Does` test/workflows/lib/v1.7.0/k8s.libsonnet` and `test/workflows/lib/v1.7.0/swagger.json` need to be checked in ?

**Comment by hougangliu** [comment]
> Does` test/workflows/lib/v1.7.0/k8s.libsonnet` and `test/workflows/lib/v1.7.0/swagger.json` need to be checked in ?

yes, workflows depends on it

**Comment by hougangliu** [comment]
> Thanks @hougangliu for this. qq for you and @yuzisun - given we are moving to Kustomize for next release, what`s the point of doing things vis a vis ksonnet?

Kustomize is used for install Kubeflow, as ksonnet here is used to run kfserving CICD, all kubeflow sub-project follows this couple now

File: test/workflows/environments/releasing/params.libsonnet [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  components+: {
+    // Insert component parameter overrides here. Ex:
+    // guestbook +: {
+    // name: "guestbook-dev",
+    // replicas: params.global.replicas,
+    // },
+    workflows+: {
+      bucket: 'kubeflow-releasing-artifacts',
+      gcpCredentialsSecretName: 'gcp-credentials',
+      name: 'jlewi-tf-operator-release-403-2f58',
```
[the remark this answers is not in the document]
Comment (hougangliu) [reply test/workflows/environments/releasing/params.libsonnet]: in fact releasing here is unused now, just leave here to complete it then when we need

File: test/workflows/components/workflows.libsonnet [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+                ],
+              },
+            },  // checkout
+            $.parts(namespace, name, overrides).e2e(prow_env, bucket).buildTemplate("run-tests", helmImage, [
+              "test/scripts/run-tests.sh",
+            ]),  // run tests
+            $.parts(namespace, name, overrides).e2e(prow_env, bucket).buildTemplate("build-kfserving", testWorkerImage, [
+              "test/scripts/build-kfserving.sh",
+            ]),  // build-kfserving
+            $.parts(namespace, name, overrides).e2e(prow_env, bucket).buildTemplate("unit-test", testWorkerImage, [
```
[the remark this answers is not in the document]
Comment (hougangliu) [reply test/workflows/components/workflows.libsonnet]: here just defines the template, the L194~L213 is the order of step

**Comment by ellis-bigelow** [comment]
Thanks Hougang. One small request, can we separate the unit test fixes from the test infrastructure? Overall I think it looks fine. It's much more complicated than I want it to be, but this may be a necessary evil for now.

**Comment by hougangliu** [comment]
> Thanks Hougang. One small request, can we separate the unit test fixes from the test infrastructure? Overall I think it looks fine. It's much more complicated than I want it to be, but this may be a necessary evil for now.

I am sorry to introduce the unit test issue, but if I submit the unit test fixes and the test infrastructure in two PRs, both of them kubeflow-kfserving-presubmit will fail

**Comment by ellis-bigelow** [comment]
> > Thanks Hougang. One small request, can we separate the unit test fixes from the test infrastructure? Overall I think it looks fine. It's much more complicated than I want it to be, but this may be a necessary evil for now.
> 
> I am sorry to introduce the unit test issue, but if I submit the unit test fixes and the test infrastructure in two PRs, both of them kubeflow-kfserving-presubmit will fail

I don't understand. Surely you can submit the test changes before the presubmits are running. is HEAD currently broken?

**Comment by hougangliu** [comment]
> I don't understand. Surely you can submit the test changes before the presubmits are running. is HEAD currently broken?

No, since https://github.com/kubernetes/test-infra/pull/12466 is merged not, each PR in kfserving will trigger kubeflow-kfserving-presubmit test, which need prow_config.yaml on root repo path

**Comment by ellis-bigelow** [comment]
Understood.

**Comment by ellis-bigelow** [comment]
/lgtm
I'll let @yuzisun have the final word.

**Comment by yuzisun** [comment]
Great work! Thanks @hougangliu !
/approve

**Pull request merged**